sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -779,6 +784,37 @@ struct _pi_queue : _pi_object {
   // asked to not transfer the ownership to SYCL RT.
   bool OwnZeCommandQueue;
 
+  // Indicates if we create an event for each command or use eventless mode for
+  // the queue. The eventless mode is used for SYCL in-order queue with
+  // discard_events property. Since we don't create L0 events for some commands
+  // we guarantee in-order semantics inside command-list by using barriers and
+  // create a special event for the last barrier of the command-list to
+  // guarantee the semantics between command-lists.
+  bool EventlessMode = false;
+
+  // It helps to skip creating a special service barrier with an event in
+  // command-list in eventless mode if we do QueueFinish or piQueueRelease
+  bool SkipLastEventInEventlessMode = false;
+
+  // Indicates if the previous command was submitted into command-list of the
+  // copy engine. it is used in eventless mode to identify that we switch
+  // command-list of copy queue to command-list of compute queue or vice versa.
+  // if the switch happened then we add a special service barrier with an event
+  // into the last command-list to maintain the order between the two
+  // command-lists.
+  bool IsPrevCopyEngine = false;
+
+  // Keeps an event of last special service barrier created to ensure in-order
+  // semantics between command-lists. The plugin is a holder of this event and
+  // SYCL RT knows nothing about it. This special event will be destroyed and
+  // returned back in pool by EventRelease in resetCommandList. this is used to
+  // compare against LastCommandEvent to ensure that event-dependency will only
+  // be added between command-lists.
+  pi_event LastEventInPrevCmdList = nullptr;
```

Comment:
Why does this need to be any "special"? Why not just add a true barrier/event, have it known to whoever needs it through the `_pi_queue::LastCommandEvent` that already exists for almost the same purpose?

Reply:
That is a good point, `LastEventInPrevCmdList` was needed before, but now when some "eventless" code is inlined in `createAndRetainPiZeEventList` we can remove it and continue to use `LastCommandEvent`. I fixed it.